implementations/rust/ockam/ockam_command/src/policy/show.rs

```diff
@@ -33,10 +37,74 @@ impl ShowCommand {
     }
 
     async fn async_run(&self, ctx: &Context, opts: CommandGlobalOpts) -> miette::Result<()> {
-        let node = BackgroundNodeClient::create_to_node(ctx, &opts.state, &self.at).await?;
-        let req = Request::get(policy_path(&self.resource, &self.action));
-        let policy: Policy = node.ask(ctx, req).await?;
-        println!("{}", policy.expression());
+        ShowTui::run(ctx, opts, self.clone()).await
+    }
+}
+
+pub struct ShowTui {
+    ctx: Context,
+    opts: CommandGlobalOpts,
+    node: BackgroundNodeClient,
+    cmd: ShowCommand,
+}
+
+impl ShowTui {
+    pub async fn run(
+        ctx: &Context,
+        opts: CommandGlobalOpts,
+        cmd: ShowCommand,
+    ) -> miette::Result<()> {
+        let node = BackgroundNodeClient::create(ctx, &opts.state, &cmd.at).await?;
+        let tui = Self {
+            ctx: ctx.async_try_clone().await.into_diagnostic()?,
+            opts,
+            node,
+            cmd,
+        };
+        tui.show().await
+    }
+}
+
+#[ockam_core::async_trait]
+impl ShowCommandTui for ShowTui {
+    const ITEM_NAME: PluralTerm = PluralTerm::Policy;
+
+    fn cmd_arg_item_name(&self) -> Option<&str> {
+        Some(self.cmd.resource.as_str())
+    }
+
+    fn terminal(&self) -> Terminal<TerminalStream<Term>> {
+        self.opts.terminal.clone()
+    }
+
+    async fn get_arg_item_name_or_default(&self) -> miette::Result<String> {
+        Ok(self.cmd.resource.to_string())
+    }
+
+    async fn list_items_names(&self) -> miette::Result<Vec<String>> {
+        Ok(self
+            .node
+            .list_policies(&self.ctx, &self.cmd.resource)
+            .await?
+            .into_iter()
+            .map(|i| i.resource().to_string())
+            .collect())
+    }
+
+    async fn show_single(&self, item_name: &str) -> miette::Result<()> {
+        let policy = self
+            .node
+            .show_policy(&self.ctx, &Resource::new(item_name), &self.cmd.action)
```

Comment:
Is that going to show a policy that is global but is applied to that resource&action pair?

Reply:
The commands now have a more predictable behavior. If you query a resource, it will return the whole policy, including the global one that applies to it, if any. If you query a resource type, it will just return its policy.